lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -68,6 +68,13 @@ LogicalResult AdvancedExtensionAttr::verify(
   return success();
 }
 
+LogicalResult mlir::substrait::FixedCharType::verify(
+    llvm::function_ref<mlir::InFlightDiagnostic()> emitError, int32_t length) {
+  if (length < 1 || length > 2147483647)
```

Comment:
Where does this constant come from? Is it `2^31 - 1`? If so, use that. Also: `int32_t` doesn't hold larger values that that, so the test is always true.